### ISSUE TITLE
When executing the infer command, it will remain stuck【windows+python312】

### ISSUE THREAD

**Issue by zhaojigang** [ASLP-lab/DiffRhythm#86]
`D:\ai\voice\DiffRhythm\diffrhythm_env\python.exe D:\ai\voice\DiffRhythm\infer\infer.py --lrc-path D:\ai\voice\DiffRhythm\infer\example\eg_en.lrc --ref-prompt "classical genres, hopeful mood, piano." --audio-length 95 --repo_id ASLP-lab/DiffRhythm-base --output-dir D:\ai\voice\DiffRhythm\infer\example\output_en --chunked
Current working directory: D:\ai\voice\DiffRhythm
./g2p\sources\g2p_chinese_model\poly_bert_model.onnx
D:\ai\voice\DiffRhythm\diffrhythm_env\Lib\site-packages\torch\nn\utils\weight_norm.py:143: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
  WeightNorm.apply(module, name, dim)
The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is ignored.`

when i When executing the above command, it will remain stuck

**Comment by NZqian** [comment]
The log looks fine, is the GPU usage normal?

**Comment by zhaojigang** [comment]
Hi, I find GPU usage is abnormal

![Image](https://github.com/user-attachments/assets/af23620d-a2ca-40c5-a6c9-f5451f0173cd)

my 'conda list'
`# Name                    Version                   Build  Channel
accelerate                1.4.0                    pypi_0    pypi
aiofiles                  23.2.1                   pypi_0    pypi
aiohappyeyeballs          2.6.1                    pypi_0    pypi
aiohttp                   3.11.16                  pypi_0    pypi
aiosignal                 1.3.2                    pypi_0    pypi
annotated-types           0.7.0                    pypi_0    pypi
anyio                     4.9.0                    pypi_0    pypi
argparse                  1.4.0                    pypi_0    pypi
attrs                     25.3.0                   pypi_0    pypi
audioread                 3.0.1                    pypi_0    pypi
babel                     2.17.0                   pypi_0    pypi
beartype                  0.20.2                   pypi_0    pypi
bitsandbytes              0.45.4                   pypi_0    pypi
bzip2                     1.0.8                h2bbff1b_6
ca-certificates           2025.2.25            haa95532_0
certifi                   2025.1.31                pypi_0    pypi
cffi                      1.17.1                   pypi_0    pypi
charset-normalizer        3.4.1                    pypi_0    pypi
click                     8.1.8                    pypi_0    pypi
cn2an                     0.5.23                   pypi_0    pypi
colorama                  0.4.6                    pypi_0    pypi
coloredlogs               15.0.1                   pypi_0    pypi
configparser              7.2.0                    pypi_0    pypi
csvw                      3.5.1                    pypi_0    pypi
decorator                 5.2.1                    pypi_0    pypi
deprecated                1.2.18                   pypi_0    pypi
dlinfo                    2.0.0                    pypi_0    pypi
docker-pycreds            0.4.0                    pypi_0    pypi
easydict                  1.13                     pypi_0    pypi
einops                    0.8.1                    pypi_0    pypi
einx                      0.3.0                    pypi_0    pypi
ema-pytorch               0.7.7                    pypi_0    pypi
expat                     2.6.4                h8ddb27b_0
fastapi                   0.115.12                 pypi_0    pypi
ffmpy                     0.5.0                    pypi_0    pypi
filelock                  3.18.0                   pypi_0    pypi
flatbuffers               25.2.10                  pypi_0    pypi
frozendict                2.4.6                    pypi_0    pypi
frozenlist                1.5.0                    pypi_0    pypi
fsspec                    2025.3.2                 pypi_0    pypi
ftfy                      6.3.1                    pypi_0    pypi
gin-config                0.5.0                    pypi_0    pypi
gitdb                     4.0.12                   pypi_0    pypi
gitpython                 3.1.44                   pypi_0    pypi
gradio                    5.23.3                   pypi_0    pypi
gradio-client             1.8.0                    pypi_0    pypi
groovy                    0.1.2                    pypi_0    pypi
h11                       0.14.0                   pypi_0    pypi
httpcore                  1.0.7                    pypi_0    pypi
httpx                     0.28.1                   pypi_0    pypi
huggingface-hub           0.30.1                   pypi_0    pypi
humanfriendly             10.0                     pypi_0    pypi
idna                      3.10                     pypi_0    pypi
inflect                   7.5.0                    pypi_0    pypi
isodate                   0.7.2                    pypi_0    pypi
jaconv                    0.4.0                    pypi_0    pypi
jieba                     0.42.1                   pypi_0    pypi
jinja2                    3.1.6                    pypi_0    pypi
joblib                    1.4.2                    pypi_0    pypi
jsonschema                4.23.0                   pypi_0    pypi
jsonschema-specifications 2024.10.1                pypi_0    pypi
language-tags             1.2.0                    pypi_0    pypi
lazy-loader               0.4                      pypi_0    pypi
libffi                    3.4.4                hd77b12b_1
librosa                   0.10.2.post1             pypi_0    pypi
lightning-utilities       0.14.3                   pypi_0    pypi
llvmlite                  0.44.0                   pypi_0    pypi
loguru                    0.7.3                    pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                3.0.2                    pypi_0    pypi
mdurl                     0.1.2                    pypi_0    pypi
more-itertools            10.6.0                   pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
msgpack                   1.1.0                    pypi_0    pypi
multidict                 6.3.2                    pypi_0    pypi
muq                       0.1.0                    pypi_0    pypi
mutagen                   1.47.0                   pypi_0    pypi
networkx                  3.4.2                    pypi_0    pypi
nnaudio                   0.3.3                    pypi_0    pypi
numba                     0.61.0                   pypi_0    pypi
numpy                     2.1.3                    pypi_0    pypi
onnxruntime-gpu           1.21.0                   pypi_0    pypi
openssl                   3.0.16               h3f729d1_0
orjson                    3.10.16                  pypi_0    pypi
packaging                 24.2                     pypi_0    pypi
pandas                    2.2.3                    pypi_0    pypi
phonemizer                3.3.0                    pypi_0    pypi
pillow                    11.1.0                   pypi_0    pypi
pip                       25.0            py312haa95532_0
platformdirs              4.3.7                    pypi_0    pypi
pooch                     1.8.2                    pypi_0    pypi
prefigure                 0.0.10                   pypi_0    pypi
proces                    0.1.7                    pypi_0    pypi
propcache                 0.3.1                    pypi_0    pypi
protobuf                  5.29.4                   pypi_0    pypi
psutil                    7.0.0                    pypi_0    pypi
py3langid                 0.3.0                    pypi_0    pypi
pyarrow                   19.0.1                   pypi_0    pypi
pycparser                 2.22                     pypi_0    pypi
pydantic                  2.11.2                   pypi_0    pypi
pydantic-core             2.33.1                   pypi_0    pypi
pydub                     0.25.1                   pypi_0    pypi
pygments                  2.19.1                   pypi_0    pypi
pykakasi                  2.3.0                    pypi_0    pypi
pylance                   0.23.2                   pypi_0    pypi
pyparsing                 3.2.3                    pypi_0    pypi
pypinyin                  0.53.0                   pypi_0    pypi
pyreadline3               3.5.4                    pypi_0    pypi
python                    3.12.9               h14ffc60_0
python-dateutil           2.9.0.post0              pypi_0    pypi
python-multipart          0.0.20                   pypi_0    pypi
pytorch-lightning         2.5.1                    pypi_0    pypi
pytz                      2025.2                   pypi_0    pypi
pyyaml                    6.0.2                    pypi_0    pypi
rdflib                    7.1.4                    pypi_0    pypi
referencing               0.36.2                   pypi_0    pypi
regex                     2024.11.6                pypi_0    pypi
requests                  2.32.3                   pypi_0    pypi
rfc3986                   1.5.0                    pypi_0    pypi
rich                      14.0.0                   pypi_0    pypi
rpds-py                   0.24.0                   pypi_0    pypi
ruff                      0.11.3                   pypi_0    pypi
safehttpx                 0.1.6                    pypi_0    pypi
safetensors               0.5.3                    pypi_0    pypi
scikit-learn              1.6.1                    pypi_0    pypi
scipy                     1.15.2                   pypi_0    pypi
segments                  2.3.0                    pypi_0    pypi
semantic-version          2.10.0                   pypi_0    pypi
sentry-sdk                2.25.1                   pypi_0    pypi
setproctitle              1.3.5                    pypi_0    pypi
setuptools                75.8.0          py312haa95532_0
shellingham               1.5.4                    pypi_0    pypi
six                       1.17.0                   pypi_0    pypi
smmap                     5.0.2                    pypi_0    pypi
sniffio                   1.3.1                    pypi_0    pypi
soundfile                 0.13.1                   pypi_0    pypi
soxr                      0.5.0.post1              pypi_0    pypi
sqlite                    3.45.3               h2bbff1b_0
starlette                 0.46.1                   pypi_0    pypi
sympy                     1.13.1                   pypi_0    pypi
threadpoolctl             3.6.0                    pypi_0    pypi
tk                        8.6.14               h0416ee5_0
tokenizers                0.21.1                   pypi_0    pypi
tomlkit                   0.13.2                   pypi_0    pypi
torch                     2.6.0                    pypi_0    pypi
torchaudio                2.6.0                    pypi_0    pypi
torchdiffeq               0.2.5                    pypi_0    pypi
torchmetrics              1.7.0                    pypi_0    pypi
torchvision               0.21.0                   pypi_0    pypi
tqdm                      4.67.1                   pypi_0    pypi
transformers              4.49.0                   pypi_0    pypi
typeguard                 4.4.2                    pypi_0    pypi
typer                     0.15.2                   pypi_0    pypi
typing-extensions         4.13.1                   pypi_0    pypi
typing-inspection         0.4.0                    pypi_0    pypi
tzdata                    2025.2                   pypi_0    pypi
unidecode                 1.3.8                    pypi_0    pypi
uritemplate               4.1.1                    pypi_0    pypi
urllib3                   2.3.0                    pypi_0    pypi
uvicorn                   0.34.0                   pypi_0    pypi
vc                        14.42                haa95532_4
vs2015_runtime            14.42.34433          he0abc0d_4
wandb                     0.19.9                   pypi_0    pypi
wcwidth                   0.2.13                   pypi_0    pypi
websockets                15.0.1                   pypi_0    pypi
wheel                     0.45.1          py312haa95532_0
win32-setctime            1.2.0                    pypi_0    pypi
wrapt                     1.17.2                   pypi_0    pypi
x-clip                    0.14.4                   pypi_0    pypi
x-transformers            2.1.2                    pypi_0    pypi
xz                        5.6.4                h4754444_1
yarl                      1.18.3                   pypi_0    pypi
zlib                      1.2.13               h8cc25b3_1`

**Comment by nitinmukesh** [comment]
All 3 are installed without CUDA

torch                     2.6.0                    pypi_0    pypi
torchaudio                2.6.0                    pypi_0    pypi
torchvision               0.21.0                   pypi_0    pypi

**Comment by zhaojigang** [comment]
> All 3 are installed without CUDA, so it may be using CPU
> 
> torch 2.6.0 pypi_0 pypi torchaudio 2.6.0 pypi_0 pypi torchvision 0.21.0 pypi_0 pypi

thanks, i uninstall 3 above package, then install `pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu126` ,it works
